core/src/main/java/org/apache/accumulo/core/iterators/Combiner.java

```diff
@@ -149,6 +159,37 @@ public void next() throws IOException {
 
   private Key workKey = new Key();
 
+  private static Cache<String,Long> loggedMsgCache = CacheBuilder.newBuilder().expireAfterWrite(1, TimeUnit.HOURS).build();
```

Comment:
Would be safer to include a high-watermark on the size of the cache. Maybe 500 elements just to prevent catastrophic failure if we somehow create a bug that causes things to be added to this cache incorrectly.